vm/actor/src/builtin/miner/mod.rs

```diff
@@ -1,62 +1,87 @@
+pub use partition_state::*;
+pub use policy::*;
+pub use quantize::*;
+pub use sector_map::*;
+pub use sectors::*;
+pub use termination::*;
+pub use types::SectorOnChainInfo;
+pub use types::*;
+pub use vesting_state::*;
+
+use crate::{account::Method as AccountMethod, market::ActivateDealsParams};
+use crate::{
+    check_empty_params, is_principal, make_map, smooth::FilterEstimate, ACCOUNT_ACTOR_CODE_ID,
+    BURNT_FUNDS_ACTOR_ADDR, CALLER_TYPES_SIGNABLE, INIT_ACTOR_ADDR, REWARD_ACTOR_ADDR,
+    STORAGE_MARKET_ACTOR_ADDR, STORAGE_POWER_ACTOR_ADDR,
 };
-use crate::power::{
-    EnrollCronEventParams, Method as PowerMethod, SectorTermination, SECTOR_TERMINATION_EXPIRED,
-    SECTOR_TERMINATION_FAULTY, SECTOR_TERMINATION_MANUAL,
+use crate::{
+    market::{
+        ComputeDataCommitmentParamsRef, Method as MarketMethod, OnMinerSectorsTerminateParams,
+        OnMinerSectorsTerminateParamsRef, VerifyDealsForActivationParamsRef,
+        VerifyDealsForActivationReturn,
+    },
+    power::CurrentTotalPowerReturn,
 };
 use crate::{
-    check_empty_params, is_principal, make_map, ACCOUNT_ACTOR_CODE_ID, BURNT_FUNDS_ACTOR_ADDR,
-    CALLER_TYPES_SIGNABLE, INIT_ACTOR_ADDR, REWARD_ACTOR_ADDR, STORAGE_MARKET_ACTOR_ADDR,
-    STORAGE_POWER_ACTOR_ADDR,
+    power::{EnrollCronEventParams, Method as PowerMethod},
+    reward::ThisEpochRewardReturn,
 };
 use address::{Address, Payload, Protocol};
-use ahash::AHashSet;
 use bitfield::BitField;
 use byteorder::{BigEndian, ByteOrder};
 use cid::{multihash::Blake2b256, Cid};
 use clock::ChainEpoch;
 use crypto::DomainSeparationTag::{
-    InteractiveSealChallengeSeed, SealRandomness, WindowedPoStChallengeSeed,
+    self, InteractiveSealChallengeSeed, SealRandomness, WindowedPoStChallengeSeed,
 };
 use encoding::Cbor;
 use fil_types::{
     InteractiveSealRandomness, PoStProof, PoStRandomness, RegisteredSealProof,
-    SealRandomness as SealRandom, SealVerifyInfo, SealVerifyParams, SectorID, SectorNumber,
-    SectorSize, WindowPoStVerifyInfo,
+    SealRandomness as SealRandom, SealVerifyInfo, SealVerifyParams, SectorID, SectorInfo,
+    SectorNumber, SectorSize, WindowPoStVerifyInfo, MAX_SECTOR_NUMBER,
 };
 use ipld_amt::Amt;
 use ipld_blockstore::BlockStore;
 use num_bigint::bigint_ser::{BigIntDe, BigIntSer};
 use num_bigint::BigInt;
 use num_derive::FromPrimitive;
-use num_traits::{FromPrimitive, Zero};
+use num_traits::{FromPrimitive, Signed, Zero};
 use runtime::{ActorCode, Runtime};
 use std::collections::HashMap;
 use std::error::Error as StdError;
-use std::ops::Neg;
+use std::{cmp, iter, ops::Neg};
 use vm::{
-    actor_error, ActorError, DealID, ExitCode, MethodNum, Serialized, TokenAmount,
-    METHOD_CONSTRUCTOR, METHOD_SEND,
+    ActorError, DealID, ExitCode, MethodNum, Serialized, TokenAmount, METHOD_CONSTRUCTOR,
+    METHOD_SEND,
 };
 
-// * Updated to specs-actors commit: 9e8c0d1c40d8b41de5dc727b6791c89e14fea4a8
+// * Updated to specs-actors commit: b8a3a6ff7b15ac01f0534c47059e1c81652a61f0
```

Comment:
Do you mind updating to 0.9.3 commit? There is only a few changes, but the control address logic that is used in the other actors for this version would be more inline.

Here is the diff: https://github.com/filecoin-project/specs-actors/compare/b8a3a6ff7b15ac01f0534c47059e1c81652a61f0...f4024efad09a66e32bfeef10a2845b2b35325297

0.9.3 commit is f4024efad09a66e32bfeef10a2845b2b35325297 (if you could specify the version here also that would be so helpful for tracking)